pelican/tests/support.py

```diff
@@ -236,24 +212,103 @@ def diff_subproc(first, second):
     )
 
 
-class LoggedTestCase(unittest.TestCase):
-    """A test case that captures log messages."""
+class LogCountHandler(BufferingHandler):
+    """Capturing and counting logged messages."""
 
-    def setUp(self):
-        super().setUp()
-        self._logcount_handler = LogCountHandler()
-        logging.getLogger().addHandler(self._logcount_handler)
+    def __init__(self, capacity=1000):
+        super().__init__(capacity)
 
-    def tearDown(self):
-        logging.getLogger().removeHandler(self._logcount_handler)
-        super().tearDown()
-
-    def assertLogCountEqual(self, count=None, msg=None, **kwargs):
-        actual = self._logcount_handler.count_logs(msg=msg, **kwargs)
-        self.assertEqual(
-            actual, count,
-            msg='expected {} occurrences of {!r}, but found {}'.format(
-                count, msg, actual))
+    @classmethod
+    @contextmanager
+    def examine(cls, loggerObj):
+        """
+        Context in which a logger's propagated messages can be examined.
+
+        Yields
+        ======
+        A handle to ``LogCountHandler.assert_count`` that has been added to the
+        specified logger for the duration of the context.
+
+        The yielded caller can be used to assert whether a certain number of
+        log messages have occurred within the context.
+        """
```

Comment:
Yes, I wasn't feeling 100% with how this reads as-is.  
  
  I'm hesitant to return an instance of the handler though, as this is never necessary, and makes usage more verbose than needed.
  
  Perhaps something like this would work?
  
  ```python
  with LogCountHandler.examine(logger) as assert_count:
      perform_work()
      assert_count(...)
  ```

Reply:
I *am* trying to make it verbose. Currently the function name (`as foo`) is at the mercy of the user, and people are lazy :). I want the interface to force them use something with an `assert` in the name.